moveit_ros/planning_interface/test/plan_using_subframes.cpp

```diff
@@ -0,0 +1,323 @@
+
+  box.subframe_names[3] = "corner_2";
+  box.subframe_poses[3].position.x = .025;
+  box.subframe_poses[3].position.y = -.05;
+  box.subframe_poses[3].position.z = -.01 + z_offset_box;
+  orientation.setRPY((90.0 / 180.0 * M_PI), 0, 0);
+  box.subframe_poses[3].orientation = tf2::toMsg(orientation);
+
+  box.subframe_names[4] = "side";
+  box.subframe_poses[4].position.x = .0;
+  box.subframe_poses[4].position.y = .0;
+  box.subframe_poses[4].position.z = -.01 + z_offset_box;
+  orientation.setRPY(0, (180.0 / 180.0 * M_PI), 0);
+  box.subframe_poses[4].orientation = tf2::toMsg(orientation);
+
+  // Next, define the cylinder
+  moveit_msgs::CollisionObject cylinder;
+  cylinder.id = "cylinder";
+  cylinder.header.frame_id = "panda_hand";
+  cylinder.primitives.resize(1);
+  cylinder.primitive_poses.resize(1);
+  cylinder.primitives[0].type = box.primitives[0].CYLINDER;
+  cylinder.primitives[0].dimensions.resize(2);
+  cylinder.primitives[0].dimensions[0] = 0.06;   // height (along x)
+  cylinder.primitives[0].dimensions[1] = 0.005;  // radius
+  cylinder.primitive_poses[0].position.x = 0.0;
+  cylinder.primitive_poses[0].position.y = 0.0;
+  cylinder.primitive_poses[0].position.z = 0.0 + z_offset_cylinder;
+  orientation.setRPY(0, (90.0 / 180.0 * M_PI), 0);
+  cylinder.primitive_poses[0].orientation = tf2::toMsg(orientation);
+
+  cylinder.subframe_poses.resize(1);
+  cylinder.subframe_names.resize(1);
+  cylinder.subframe_names[0] = "tip";
+  cylinder.subframe_poses[0].position.x = 0.03;
+  cylinder.subframe_poses[0].position.y = 0.0;
+  cylinder.subframe_poses[0].position.z = 0.0 + z_offset_cylinder;
+  orientation.setRPY(0, (90.0 / 180.0 * M_PI), 0);
+  cylinder.subframe_poses[0].orientation = tf2::toMsg(orientation);
+
+  // Lastly, the objects are published to the PlanningScene. In this tutorial, we publish a box and a cylinder.
+  box.operation = moveit_msgs::CollisionObject::ADD;
+  cylinder.operation = moveit_msgs::CollisionObject::ADD;
+  std::vector<moveit_msgs::CollisionObject> collision_objects = { box, cylinder };
+  planning_scene_interface.applyCollisionObjects(collision_objects);
+}
+
+// Helper function to convert pose message to eigen Affine3d
+void poseMsgToEigen(Eigen::Affine3d& left, const geometry_msgs::Pose& right)
+{
+  left = Eigen::Translation3d(right.position.x, right.position.y, right.position.z) *
+         Eigen::Quaterniond(right.orientation.w, right.orientation.x, right.orientation.y, right.orientation.z);
+}
+
+// Helper function for finding the index of a value in a list of strings
+bool findStringIndex(int* index, const std::vector<std::string>& list, const std::string& value)
+{
+  for (size_t i = 0; i < list.size(); ++i)
+  {
+    if (list[i] == value)
+    {
+      *index = i;
+      return true;
+    }
+  }
+  return false;
+}
+
+// Test method to test if the cylinder tip at near the box subrame_name meantioned
+void testAtSubframe(const std::string& subframe_name,
```

Comment:
I am always for much more verbose function names. I would call this "isCylinderTipAtBoxSubframe" or something similar.